portal-web/docroot/html/taglib/ui/user_name_fields/page.jsp

```diff
@@ -21,18 +21,22 @@
 
 </aui:select>
 
-<aui:script require="users-admin-web/js/UserNameFields.es as UserNameFields">
-	var component = Liferay.component(
-		'<portlet:namespace />UserNameFields',
-		new UserNameFields.default(
-			{
-				baseURL: '<%= HtmlUtil.escapeJS(themeDisplay.getURLCurrent()) %>',
-				formNode: <portlet:namespace />fm,
-				languageIdSelectNode: '#<portlet:namespace />languageId',
-				portletNamespace: '<portlet:namespace />',
-				userNameFieldsNode: <portlet:namespace />userNameFields
-			}
-		)
+<aui:script>
+	import('/o/users-admin-web/__liferay__/index.js').then(
+		({UserNameFields}) => {
```

Comment:
this looks a new pattern. 
On a first thought, looks that a -web module might want to import some JS provided by itself. In other cases we solved this with <liferay-ui:component>
I think we don't have to spread this pattern, however, what could we offer if more cases like this show up? what prevents us from solving this case like the others?

Reply:
It's not necessarily a new pattern. Just a standard dynamic inport. Since this is under portal-web, we don't have a build process and so we don't use npm-scripts. By using the dynamic import, it helps us circumvent that technical hurdle.

If other cases come up, but under `./modules`, we can use our build tooling